pandas/core/indexes/category.py

```diff
@@ -696,9 +696,11 @@ def get_indexer_non_unique(self, target):
 
     @Appender(_index_shared_docs["_convert_scalar_indexer"])
     def _convert_scalar_indexer(self, key, kind=None):
-        if self.categories._defer_to_indexing:
-            return self.categories._convert_scalar_indexer(key, kind=kind)
-
+        if kind == "loc" or self.categories._defer_to_indexing:
```

Comment:
we might not need the _defer_to_indexing any longer, as I think the only way to get here is kind=='loc'